File: codingjs/gHappy.js
function gHappy(str){
  const customeStr = 'x' + str + 'x';
  return gHappyHelper(customeStr, true);
}
function gHappyHelper(str, happy) {
  if (str.length < 3) return happy;
  if (/gg/.test(str.substring(0, 3))) {
    return gHappyHelper(str.substring(3), true);
  }
  if (/[^g]g[^g]/.test(str.substring(0, 3))) {
    return false;
  }
  return gHappyHelper(str.substring(1), happy);
}

File: codingbat/gHappy.java
public boolean gHappy(String str) {
  boolean happy = true;
  for (int i = 0; i < str.length(); i++) {
    if (str.charAt(i) == 'g') {
      if (i > 0 && str.charAt(i - 1) == 'g') happy = true;
      else if (i < str.length() - 1 && str.charAt(i + 1) == 'g') happy = true;
      else happy = false;
    }
  }
  return happy;
}

